### PR TITLE
Feature 587: only refresh affected weekdays when making changes

### LINKED ISSUE #587: Moving activities in a weekplan should not cause the whole weekplan to reload
**Is your feature request related to a problem? Please describe.**
When moving activities, the whole weekplan reloads. All cancelations and completions are removed and added again. It is especially bad if pictogram text is visible, as it causes the whole plan to jump around (See video).

**Describe the solution you'd like**
The weekplan should not visibly go through a reload as seen in the video and described above. It should just move the activity.

**Describe alternatives you've considered**
Hide the whole weekplan until the reload is finished.

**Additional context**

Related to aau-giraf/web-api#19

ezgif-6-e9f84d0100ae

### PR DESCRIPTION
**Pull request by AdrianPlesner**
Closes #587

### CODE REVIEW

**Comment by Determinatoro** [comment]
Please use the following checklist, @social2dk. Feel free to check it off as you go along.

**Assets**
- [x] The ``environments.json`` file has the following default values:
```
{
  "SERVER_HOST": "https://srv.giraf.cs.aau.dk/DEV/API",
  "DEBUG": true,
  "USERNAME": "Graatand",
  "PASSWORD": "password"
}
```

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [ ] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [x] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [x] All functionality is covered by tests
- [x] The tests has sensible names
- [x] By reading the tests you know what they cover
- [x] The tests cover sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Comment by Determinatoro** [comment]
Please use the following checklist, @Riasalit. Feel free to check it off as you go along.

**Assets**
- [x] The ``environments.json`` file has the following default values:
```
{
  "SERVER_HOST": "https://srv.giraf.cs.aau.dk/DEV/API",
  "DEBUG": true,
  "USERNAME": "Graatand",
  "PASSWORD": "password"
}
```

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [x] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [x] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [x] All functionality is covered by tests
- [x] The tests has sensible names
- [x] By reading the tests you know what they cover
- [x] The tests cover sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Comment by AdrianPlesner** [comment]
Vær opmærksom på at dette er linket sammen med aau-giraf/api_client#81 og aau-giraf/web-api#19 og det er derfor selvfølgelig nødvendigt at have de tilhørende branches af de repositories for at der virker ordentligt.

**Comment by AdrianPlesner** [comment]
CI fejler og det er fordi at den tilhørende branch på api-client ikke er merged endnu, men jeg vil ikke pushe en pub-spec der peger på andet en develop. Vær venlig at review alligevel. ^_^

**Comment by Riasalit** [comment]
Når jeg flytter en ugeplan for først egang bliver den ikke flyttet

Flytter jeg den samme igen forsvinder den originale ikke, men jeg kan duplikere den ind i andre dage

File: pubspec.lock [hunk 1]
```diff
@@ -5,8 +5,8 @@ packages:
     dependency: "direct main"
     description:
       path: "."
-      ref: develop
-      resolved-ref: a2dbd27fc9743f9994e2434f503d3645aef5a034
+      ref: "feature/81"
```
Comment (Riasalit) [inline pubspec.lock]: Gætter på at det ikke var meningen at pubspec skulle med?

File: lib/widgets/weekplanner_choiceboard_selector.dart [hunk 2]
```diff
@@ -19,15 +20,15 @@ import 'giraf_confirm_dialog.dart';
...
   ///Constructor
   WeekplannerChoiceboardSelector(this._activity, this._activityBloc, this._user,
-      this._weekplanBloc, this._week) {
+      this._weekplanBloc, this._weekday) {
     _activityBloc.load(_activity, _user);
   }
 
   final ActivityModel _activity;
 
   final WeekplanBloc _weekplanBloc;
```
Comment (Riasalit) [inline lib/widgets/weekplanner_choiceboard_selector.dart]: er den her brugt?

File: lib/widgets/weekplanner_choiceboard_selector.dart [hunk 3]
```diff
@@ -19,15 +20,15 @@ import 'giraf_confirm_dialog.dart';
...
 
-  final WeekModel _week;
+  final WeekdayModel _weekday;
```
Comment (Riasalit) [inline lib/widgets/weekplanner_choiceboard_selector.dart]: og den her?

File: lib/widgets/weekplanner_choiceboard_selector.dart (hunk 2 above)
Comment (Riasalit) [reply lib/widgets/weekplanner_choiceboard_selector.dart]: min linter klager over unused imports

Comment (Rosmarium) [reply lib/widgets/weekplanner_choiceboard_selector.dart]: Hej, vi kigger lige på det :)

**Review by Riasalit — CHANGES_REQUESTED** [review]
Flytter jeg den samme igen forsvinder den originale ikke, men jeg kan duplikere den ind i andre dage

**Comment by AdrianPlesner** [comment]
> Flytter jeg den samme igen forsvinder den originale ikke, men jeg kan duplikere den ind i andre dage

@Riasalit Prøv at se om det ikke er bedre nu

**Review by Riasalit — DISMISSED** [review]

**Comment by Riasalit** [comment]
Merge conflicts

**Comment by LivHolm** [comment]
@AdrianPlesner husk at resolve conflicts ^^

**Comment by AdrianPlesner** [comment]
Jeg er i gang. ^_^

**Comment by AdrianPlesner** [comment]
2 sek der var problemer med den merge der. 🙃

**Comment by Riasalit** [comment]
image
eehhhhh
får en error hvis jeg trykker på noget?

**Comment by Riasalit** [comment]
> 
> 
> image
> eehhhhh
> får en error hvis jeg trykker på noget?

Muligvis bare mig der mangler at gendanne databasen?

**Comment by AdrianPlesner** [comment]
@Riasalit Ja der mangler titler på dine activiteter

**Comment by Riasalit** [comment]
det hjalp... men 
image

**Comment by Riasalit** [comment]
kan se at 5 af billederne i min database er corrupte

**Comment by AdrianPlesner** [comment]
aha

**Comment by Riasalit** [comment]
det var ikke problemet... de starte med at loade ind rigtigt. men så bliver de alle røde og teksten forsvinder fra dem

**Comment by AdrianPlesner** [comment]
Har de en titel?

**Comment by Riasalit** [comment]
nope

**Comment by AdrianPlesner** [comment]
Så er det problemet, giv følgende query til din database og se om det ikke hjælper:
```
with ackeynam as (select min(Pictograms.Title) as name, Activities.Key as acKey
from Pictograms, PictogramRelations, Activities
where Pictograms.id = PictogramRelations.PictogramId AND Activities.Key = PictogramRelations.ActivityId
GROUP BY Activities.Key)
update Activities
set Title = (
    select name 
    from ackeynam
    where acKey = Activities.Key
    );
```

**Comment by Riasalit** [comment]
såååå. det virkede men hvis jeg prøver at indsætte  en ny, så kan jeg vælge et piktogram og så sætter den bare instantly den ind uden at lade mig give den en tittel og så bliver den rød

**Comment by AdrianPlesner** [comment]
> såååå. det virkede men hvis jeg prøver at indsætte en ny, så kan jeg vælge et piktogram og så sætter den bare instantly den ind uden at lade mig give den en tittel og så bliver den rød

Reee

**Comment by Riasalit** [comment]
jeg kan så derfefter trykke på den røde aktivitet og give den en tittel.. men burde den menu ikke komme frem af sig selv?

**Comment by AdrianPlesner** [comment]
> jeg kan så derfefter trykke på den røde aktivitet og give den en tittel.. men burde den menu ikke komme frem af sig selv?

Jo det burde den helt sikkert. *suk*

**Comment by Riasalit** [comment]
Det skal lige siges  at jeg bruger develop på api-client og web-api, ved ikke om det er det rigtige?

**Comment by AdrianPlesner** [comment]
> Det skal lige siges at jeg bruger develop på api-client og web-api, ved ikke om det er det rigtige?

Jeg tror det er rigtigt nok, de er vidst merged

**Comment by AdrianPlesner** [comment]
@Riasalit Så kører vi!

**Comment by Riasalit** [comment]
nice, men øøøøh, environments.json ^^

**Comment by Riasalit** [comment]
altså... hvis jeg allerede har flere der bruger samme piktogram men med flere navne tager den lidt en tilfeldig. men tror at det er okay?

File: assets/environments.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "SERVER_HOST": "https://srv.giraf.cs.aau.dk/DEV/API",
+  "SERVER_HOST": "http://localhost:5000",
```
Comment (Riasalit) [inline assets/environments.json]: this bad. men tror det er det sidste ^^

**Review by Riasalit — CHANGES_REQUESTED** [review]

File: assets/environments.json (hunk 4 above)
Comment (AdrianPlesner) [reply assets/environments.json]: oops

**Comment by AdrianPlesner** [comment]
> altså... hvis jeg allerede har flere der bruger samme piktogram men med flere navne tager den lidt en tilfeldig. men tror at det er okay?

Jeg er ikke helt sikker på at jeg ved hvad du mener

**Comment by Riasalit** [comment]
> 
> 
> > altså... hvis jeg allerede har flere der bruger samme piktogram men med flere navne tager den lidt en tilfeldig. men tror at det er okay?
> 
> Jeg er ikke helt sikker på at jeg ved hvad du mener

Hvis jeg laver en ny aktivitet med piktogrammet "Gul" laver den en aktivitet der hedder Gul. men hvis jeg så laver den om til at have titlen "Middagsmad" og så laver en ny aktivitet med piktogrammet "Gul" kommer den også til at hedde "Middagsmad" i ugeplanen. 

Men igen. ved ikke om det overhovedet er et problem?

**Comment by AdrianPlesner** [comment]
>Hvis jeg laver en ny aktivitet med piktogrammet "Gul" laver den en aktivitet der hedder Gul. men hvis jeg så laver den om til at have titlen "Middagsmad" og så laver en ny aktivitet med piktogrammet "Gul" kommer den også til at hedde "Middagsmad" i ugeplanen.
Men igen. ved ikke om det overhovedet er et problem?

Nej det er lige præcis meningen med #266

**Review by Riasalit — APPROVED** [review]

**Review by social2dk — APPROVED** [review]

**Pull request merged**